File: extensions/use_c_and_nim/code.nim
import std/strformat

proc f(p1: ptr uint8, p2: ptr UncheckedArray[uint8]): int {.exportc, cdecl, dynlib.} =
  echo fmt"addr: 0x{cast[uint64](p2):x}"
  let delta = cast[uint64](p2) - cast[uint64](p1)
  echo fmt"delta: 0x{delta:x}"
  for i in 0..4:
    echo fmt"array[{i}] = {p2[i]}"
    p2[i] += 1

File: extensions/use_c_and_nim/code.c
#include <stdio.h>

int f(unsigned char *p1, unsigned char *p2) {
	printf("addr: %p\n", p2);
	printf("delta: %p\n", p2 - p1);
	unsigned char *p = p2;
	for (int i = 0; i < 5; i++)
		printf("array[%d] = %d\n", i, p[i]++);
}
